### DISCUSSION TITLE
Rich version history or latest release?

### DISCUSSION THREAD

**Discussion by mdanish-kh** [microsoft/winget-pkgs#66854] [Q&A] [answered]
This may be a trivial question, but I'm seeing gaps in releases more frequently so that's why I just needed to confirm this.

Recently, I updated some packages and noticed that there was a gap in release versions from the last updated manifest to the latest release. One example being [NZXT.CAM](https://github.com/microsoft/winget-pkgs/tree/master/manifests/n/NZXT/CAM) , the last updated one being `4.34.3` and currently the latest release being `4.37.1`. My question is if it is preferable to keep a rich version history and include all releases (in this case `4.35.0`, `4.36.0`, `4.36.1`, `4.37.0`)  where possible or just bother with the latest release?

**Comment by OfficialEsco** [comment, marked as answer]
As you can see i've been maintaining the NZXT CAM Package, however its very difficult to keep a track off since the only way i've found reliable is to download and install NZXT CAM from https://nzxt-app.nzxt.com/NZXT-CAM-Setup.exe
Then note down the version and change the URL https://nzxt-app.nzxt.com/NZXT%20CAM%20Setup%204.37.1.exe

I do however believe NZXT CAM also do not always release every version, they also don't always release patch notes for every release.

The reason behind the 2 month gap is because i've taken a burnout break AND i do not use NZXT CAM.
The best way to keep a package up to date is to have a user of the software update it once the update rolls out.

[NordVPN](https://github.com/microsoft/winget-pkgs/tree/master/manifests/n/NordVPN/NordVPN) also have the same issue as NZXT CAM (Which i also do not use but maintain)

These could be automated once we get some more advanced automation (Currently only GitHub releases are highly automated)

**Reply by mdanish-kh** [reply]
> 

I understand it is quite difficult to maintain each and every package and looking at your history of contributions and the number of packages you maintain I think you have done more than an amazing job!

I have just started to contribute to this repo as I found it like 15-20 days ago and I'm a beginner when it comes to contributing to open source projects like these. I am just asking that if I see such a case, is it okay that I add old releases as well? I don't want it to seem like I'm flooding the repo with PRs and making life hard for moderators, especially making PRs for those versions that are not the latest release and may not matter that much.

**Reply by OfficialEsco** [reply]
Ahh right i forgot to answer the question 🤦‍♂️
Quick answer; Feel free to add old versions (major and/or up to the 10 latest versions)

Long Answer:
I guess it depends on the software, we've done some cleanup on some packages which release new versions VERY OFTEN and its a mixed bag, one issue with keeping ALL versions is that the repo becomes huge for the contributors when forking, however we noticed it didn't really help..
On the the client side it makes sense to have a huge repo of old versions. In some cases we've kept the latest Patch version 1.3.X which i believe should cover the use case of old versions

In theory, feel free to add any versions you feel would be useful, but if the software force autoupdates there is no reason to add a bunch, those who REALLY want a old version will find a way to block the update and most likely install it manually anyways

IMO 99.9% of the userbase will use WinGet to install the LATEST version instead of a specific version which can find with 
```
❯ winget show mozilla.firefox --versions
Found Mozilla Firefox [Mozilla.Firefox]
Version
-------
102.0.1
102.0
101.0.1
101.0
100.0.2
100.0.1
100.0
99.0.1
99.0
98.0.2
98.0.1
98.0
97.0.2
97.0.1
97.0
96.0.3
96.0.2
96.0.1
96.0
```

**Comment by russellbanks** [comment]
If a package is outdated and only has one version in it's folder (someone submitted it but it was never updated, for example), I tend to let yamlcreate do its job and then delete the previous version (by deleting the folder of the previous version) manually. You'd have to judge this for yourself with each individual package whether it's necessary to do but otherwise you can end up with arbitrarily old packages that I can almost guarantee no one will install.

This can also be the case if there are more versions in a package but they are months or even years old (so it's very unlikely that anyone would be installing those old ones) so I would just delete the previous version here as well.

**Reply by vedantmgoyal9** [reply]
+1. Don't know why my reaction is not getting added :-(
